vm/systemSmartContracts/delegationManager.go

```diff
@@ -214,6 +219,7 @@ func (d *delegationManager) createNewDelegationContract(args *vmcommon.ContractC
 		return vmcommon.UserError
 	}
 
+	//TODO: If saveDelegationContractList will fail, then the last saves done will remain saved. Also SetStorage is already done here. Is this ok?
```

Comment:
nothing will remain. Any error invalidates all previous changes. save is done in a cache - cache is invalidated if returnCode != vmcommon.Ok

Reply:
ok